### PR TITLE
Add test vectors for trieProof verify

### PR DESCRIPTION
**Pull request by ed255**
Add 15 proof vectors to test the trieProof verify function.  Two of the test are of exclusion.

### CODE REVIEW

File: contracts/test/trie-proof.sol [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+// This contract is for testing. It exposes internal methods of
+// TrieProof lib so that we can test them.
+//
+// *************************************************************
+// *** Never deploy this contract!                           ***
+// *************************************************************
+
+contract TrieProofTest {
+    // using TrieProof for bytes;
```
Comment (jordipainan) [inline contracts/test/trie-proof.sol:15]: Any reason to keep this line commented ? If not used I am in favor of removing it :)

File: test/builders/token-storage-proof.ts [hunk 2]
```diff
@@ -18,3 +18,21 @@ export default class TokenStorageProofBuilder {
         return contractInstance.connect(this.deployAccount.wallet) as Contract & Erc20StorageProofContractMethods
     }
 }
+
+import { abi as trieProofTestAbi, bytecode as trieProofTestByteCode } from "../../build/test/trie-proof.json"
```
Comment (jordipainan) [inline test/builders/token-storage-proof.ts:22]: Do you mind to move the imports to the top of the file and put them together with the others ?

File: test/contracts/token-storage-proof.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
-import { expect } from "chai"
-import { BigNumber, Contract, ContractFactory, ContractTransaction } from "ethers"
+import { expect, assert } from "chai"
```
Comment (jordipainan) [inline test/contracts/token-storage-proof.ts:1]: `assert` is imported but not used, can be deleted :)

File: test/contracts/token-storage-proof.ts [hunk 4]
```diff
@@ -144,4 +144,194 @@ describe("TokenStorageProof contract", () => {
         expect(result.tokenAddress).to.equal(dummyTokenInstance.address)
         expect(result.balanceMappingPosition).to.be.deep.equal(BigNumber.from(5))
     })
-})
\ No newline at end of file
+})
+
+import { abi as trieProofTestAbi, bytecode as trieProofTestByteCode } from "../../build/test/trie-proof.json"
```
Comment (jordipainan) [inline test/contracts/token-storage-proof.ts:149]: Same as commented before, do you mind to move the imports to the top of the file and put them together with the others ? Thank you :)

File: test/contracts/token-storage-proof.ts [hunk 5]
```diff
@@ -144,4 +144,194 @@ describe("TokenStorageProof contract", () => {
...
+            if (p.verify) {
+                expect(value).to.equal(p.value)
+            } else {
+                expect(value).to.not.equal(p.value)
+            }
+        }
+    })
+})
+
+const testProofs = [
```
Comment (jordipainan) [inline test/contracts/token-storage-proof.ts:192]: Does it make sense for you to move this vectors on a separate file? Maybe test/vectors or similar. The reason is that maybe in the future we want to add more vectors for another thing and the code will be cleaner and more structured.
What do you think? Thank you

**Comment by jordipainan** [comment]
I'm going to merge the other PR's you can rebase from main when changes applied :)

**Comment by ed255** [comment]
> I'm going to merge the other PR's you can rebase from main when changes applied :)

I've applied the requested changes and rebased from main :)

**Pull request merged**